### PR TITLE
refactor(chart SCIM): migrate to a proper chart + add doc

### PR DESCRIPTION
**Pull request by RiceBowlJr**
Overall chart cleanup.
Fixes the mismatch between service and pod/container port (`8443` is not exposed on the pod, everything must be redirected to `8080` port).
Add a minimal doc (values array generated with [`helm-doc`](https://github.com/norwoodj/helm-docs) since I'm too lazy.
Allow more installations flexibility (usage of different storage classes, of ingresses instead of service type `LoadBalancer`, ...).

### CODE REVIEW

File: op-scim-bridge/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  {{- end }}
+  rules:
+    {{- range .Values.ingress.hosts }}
+    - host: {{ .host | quote }}
+      http:
+        paths:
+          {{- range .paths }}
+          - path: {{ . }}
+            backend:
+              serviceName: {{ $fullName }}-bridge
```
Comment (lucchmielowski) [inline op-scim-bridge/templates/ingress.yaml:37]: On peut enlever le `-bridge` à la fin non ? (ou remove le `-bridge` à la fin du nom du chart 🤷)

File: op-scim-bridge/templates/service.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "op-scim-bridge.fullname" . }}-bridge
```
Comment (lucchmielowski) [inline op-scim-bridge/templates/service.yaml:4]: (ne pas oublier de remove -bridge ici aussi si besoin

**Review by andronux — APPROVED** [review]
LGTM 👍 

quick remark: is there any prometheus metrics we can monitor?

File: op-scim-bridge/templates/tests/test-connection.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  labels:
+    {{- include "op-scim-bridge.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+    - name: wget
+      image: busybox
+      command: ['wget']
+      args: ['{{ include "op-scim-bridge.fullname" . }}:{{ .Values.service.port }}']
```
Comment (lucchmielowski) [inline op-scim-bridge/templates/tests/test-connection.yaml:14]: Peut être que l'on devrait tester le NS et pas le service ? 🤔

**Review by lucchmielowski — APPROVED** [review]
Review in Franglais but 👍

File: op-scim-bridge/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            - "/op-scim/op-scim"
+          args:
+            - "--redis-host={{ .Release.Name }}-redis-master"
+            - "--session={{ .Values.scim.userHome }}/{{ .Values.scim.sessionFile }}"
+            - "--port=8080"
+          ports:
+            - name: http-app
+              containerPort: 8080
+              protocol: TCP
+            - name: https-app
```
[the remark this answers is not in the document]
Comment (RiceBowlJr) [reply op-scim-bridge/templates/deployment.yaml:59]: We don't know 😛 
Removing it

File: op-scim-bridge/Chart.yaml [hunk 5]
```diff
@@ -9,9 +9,13 @@ keywords:
   - "SCIM"
   - "1Password"
 home: https://support.1password.com/scim
-maintainers: # (optional)
+maintainers:
   - name: 1Password Platform Integrations Team
```
[the remark this answers is not in the document]
Comment (RiceBowlJr) [reply op-scim-bridge/Chart.yaml:13]: Maintainers <> Contributors 😜

File: op-scim-bridge/values.yaml [hunk 6]
```diff
@@ -1,7 +1,104 @@
...
+ingress:
+  # -- Enable or not Ingress
+  enabled: false
+  # -- Additionnal annotations for the Ingress
+  annotations: {}
+    # kubernetes.io/ingress.class: nginx
+    # nginx.ingress.kubernetes.io/ssl-redirect: "false"
+    # kubernetes.io/tls-acme: "true"
+  # -- Hosts configuration for the ingress
+  hosts:
```
[the remark this answers is not in the document]
Comment (RiceBowlJr) [reply op-scim-bridge/values.yaml:64]: It is not created by default so I put this so it is easily usable:
```yaml
ingress:
  # -- Enable or not Ingress
  enabled: false
```

File: op-scim-bridge/templates/ingress.yaml (hunk 1 above)
Comment (RiceBowlJr) [reply op-scim-bridge/templates/ingress.yaml:37]: Indeed

Comment (RiceBowlJr) [reply op-scim-bridge/templates/ingress.yaml:37]: But I love bridges

File: op-scim-bridge/templates/tests/test-connection.yaml (hunk 3 above)
Comment (RiceBowlJr) [reply op-scim-bridge/templates/tests/test-connection.yaml:14]: Not if NS is not used (cf `service.type=LoadBalancer` qui est le default.

**Comment by RiceBowlJr** [comment]
> LGTM 👍
> 
> quick remark: is there any prometheus metrics we can monitor?

¯\_(ツ)_/¯
This is a 1Password software and does not seems to be open source. Does not seems to be some doc about this either.
[The doc](https://support.1password.com/scim/#step-2-deploy-the-scim-bridge) 🤣:
> The SCIM bridge writes to standard output (stdout) for easy log collection.

File: op-scim-bridge/templates/tests/test-connection.yaml (hunk 3 above)
Comment (RiceBowlJr) [reply op-scim-bridge/templates/tests/test-connection.yaml:14]: Resolving this :) since it depends on the configuration.

**Pull request merged**